### ISSUE TITLE
[BUG] `geo2rdr did not converge` for GMAO - think it's about Availability?

### ISSUE THREAD

**Issue by cmarshak** [dbekaert/RAiDER#488]
**Describe the bug**
This is related to hyp3 deployment and the current script to update GUNWs.

Could be potentially very simple, i.e. due to the fact that the reference date is `20230106` and GMAO is not available. If such is the case, may be helpful to create slightly better error message here.

Here is the job dict:

```
{'job_id': '5225cd05-0869-4031-a697-15778a29635f',
  'job_type': 'INSAR_ISCE_TEST',
  'request_time': '2023-02-13T22:40:20+00:00',
  'status_code': 'FAILED',
  'user_id': 'cmarshak',
  'name': 'Tibet_48_beta_gmao',
  'job_parameters': {'estimate_ionosphere_delay': True,
   'frame_id': 7420,
   'granules': ['S1A_IW_SLC__1SDV_20230106T235934_20230107T000001_046670_059804_D8E3',
    'S1A_IW_SLC__1SDV_20230106T235959_20230107T000026_046670_059804_74DB'],
   'secondary_granules': ['S1A_IW_SLC__1SDV_20221213T235935_20221214T000002_046320_058C2F_7DFC',
    'S1A_IW_SLC__1SDV_20221214T000000_20221214T000027_046320_058C2F_9AD4'],
   'weather_model': 'GMAO'},
  'logs': ['https://hyp3-a19-jpl-contentbucket-1wfnatpznlg8b.s3.us-west-2.amazonaws.com/5225cd05-0869-4031-a697-15778a29635f/5225cd05-0869-4031-a697-15778a29635f.log'],
  'expiration_time': '2023-08-14T00:00:00+00:00',
  'processing_times': [8891.189, 23.475]}
```

The log (as above) is here: https://hyp3-a19-jpl-contentbucket-1wfnatpznlg8b.s3.us-west-2.amazonaws.com/5225cd05-0869-4031-a697-15778a29635f/5225cd05-0869-4031-a697-15778a29635f.log

The GUNW is here (without tropo layer): https://hyp3-a19-jpl-contentbucket-1wfnatpznlg8b.s3.us-west-2.amazonaws.com/5225cd05-0869-4031-a697-15778a29635f/S1-GUNW-D-R-048-tops-20230106_20221214-000000-00090E_00040N-PP-c254-v2_0_6.nc

So using the GUNW, this script could be re-run.

**To Reproduce**

```
raider.py ++process calcDelaysGUNW --model GMAO <GUNW_ID>
```
See https://github.com/ASFHyP3/hyp3/issues/1416 for exactly how raider is called.

**Comment by cmarshak** [comment]
Here are the failed jobs for reference (there are 3 / 18 submitted):

```
[{'job_id': '5225cd05-0869-4031-a697-15778a29635f',
  'job_type': 'INSAR_ISCE_TEST',
  'request_time': '2023-02-13T22:40:20+00:00',
  'status_code': 'FAILED',
  'user_id': 'cmarshak',
  'name': 'Tibet_48_beta_gmao',
  'job_parameters': {'estimate_ionosphere_delay': True,
   'frame_id': 7420,
   'granules': ['S1A_IW_SLC__1SDV_20230106T235934_20230107T000001_046670_059804_D8E3',
    'S1A_IW_SLC__1SDV_20230106T235959_20230107T000026_046670_059804_74DB'],
   'secondary_granules': ['S1A_IW_SLC__1SDV_20221213T235935_20221214T000002_046320_058C2F_7DFC',
    'S1A_IW_SLC__1SDV_20221214T000000_20221214T000027_046320_058C2F_9AD4'],
   'weather_model': 'GMAO'},
  'logs': ['https://hyp3-a19-jpl-contentbucket-1wfnatpznlg8b.s3.us-west-2.amazonaws.com/5225cd05-0869-4031-a697-15778a29635f/5225cd05-0869-4031-a697-15778a29635f.log'],
  'expiration_time': '2023-08-14T00:00:00+00:00',
  'processing_times': [8891.189, 23.475]},
 {'job_id': '14dae435-36fb-4797-9ebd-250434d3d25c',
  'job_type': 'INSAR_ISCE_TEST',
  'request_time': '2023-02-13T22:40:20+00:00',
  'status_code': 'FAILED',
  'user_id': 'cmarshak',
  'name': 'Tibet_48_beta_gmao',
  'job_parameters': {'estimate_ionosphere_delay': True,
   'frame_id': 7420,
   'granules': ['S1A_IW_SLC__1SDV_20221225T235935_20221226T000002_046495_059224_A539',
    'S1A_IW_SLC__1SDV_20221225T235959_20221226T000026_046495_059224_1CFC'],
   'secondary_granules': ['S1A_IW_SLC__1SDV_20221213T235935_20221214T000002_046320_058C2F_7DFC',
    'S1A_IW_SLC__1SDV_20221214T000000_20221214T000027_046320_058C2F_9AD4'],
   'weather_model': 'GMAO'},
  'logs': ['https://hyp3-a19-jpl-contentbucket-1wfnatpznlg8b.s3.us-west-2.amazonaws.com/14dae435-36fb-4797-9ebd-250434d3d25c/14dae435-36fb-4797-9ebd-250434d3d25c.log'],
  'expiration_time': '2023-08-14T00:00:00+00:00',
  'processing_times': [8888.974, 23.301]},
 {'job_id': 'a3162b13-8591-4cd4-93e2-a02b56100d04',
  'job_type': 'INSAR_ISCE_TEST',
  'request_time': '2023-02-13T22:40:20+00:00',
  'status_code': 'FAILED',
  'user_id': 'cmarshak',
  'name': 'Tibet_48_beta_gmao',
  'job_parameters': {'estimate_ionosphere_delay': True,
   'frame_id': 7420,
   'granules': ['S1A_IW_SLC__1SDV_20230106T235934_20230107T000001_046670_059804_D8E3',
    'S1A_IW_SLC__1SDV_20230106T235959_20230107T000026_046670_059804_74DB'],
   'secondary_granules': ['S1A_IW_SLC__1SDV_20221225T235935_20221226T000002_046495_059224_A539',
    'S1A_IW_SLC__1SDV_20221225T235959_20221226T000026_046495_059224_1CFC'],
   'weather_model': 'GMAO'},
  'logs': ['https://hyp3-a19-jpl-contentbucket-1wfnatpznlg8b.s3.us-west-2.amazonaws.com/a3162b13-8591-4cd4-93e2-a02b56100d04/a3162b13-8591-4cd4-93e2-a02b56100d04.log'],
  'expiration_time': '2023-08-14T00:00:00+00:00',
  'processing_times': [8891.929, 23.342]}]
```

**Comment by dbekaert** [comment]
I think the orbits and SLC seems to overlap.
e.g. Compare the SLCs with the validity of the orbit:
 'secondary_granules': ['S1A_IW_SLC__1SDV_**20221213T235935**_20221214T000002_046320_058C2F_7DFC',
    'S1A_IW_SLC__1SDV_20221214T000000_20221214T000027_046320_058C2F_9AD4'],
with:
S1A_OPER_AUX_POEORB_OPOD_20230103T081835_V**20221213T225942**_20221215T005942.EOF

@jlmaurer we might need to have a debug look at it?

**Comment by bbuzz31** [comment]
Yeah, this is more nefarious than the availability unfortunately

**Comment by bbuzz31** [comment]
This bug is indeed related to the padding parameter (which slices the state vectors in the orbit file around the GUNW reference UTC time).

If extra padding is used (>41 minutes; the default is 10 minutes) then geo2rdr converges. 
It's not clear why this is needed: the reference time at 00:00:00 UTC, so adding padding of +/- 10 minutes is intuitively sufficient, as the total SLCs range is 23:59:34 to 00:00:27.
And if this extra padding is used, previously converging locations fail (e.g., test_scenario_3.py)

@piyushrpt could you help us understand what's happening here?

**Comment by cmarshak** [comment]
I specifically selected this sample due to its midnight crossing ;) FYI - Thanks, Rob!

**Comment by bbuzz31** [comment]
@cmarshak yeah this may have something to do with it... did any of the 15 that succeeded have a midnight crossing?

**Comment by piyushrpt** [comment]
If 10 minutes around the reference time in insufficient, I would check the source granule. With SAR, time is same as spatial extent - i.e, where the satellite is looking. If it fails to converge with the right time window - it probably means that the assigned reference time is off or that when building the orbit data structure wrong times are being assigned to the state vector.

**Comment by cmarshak** [comment]
Thank youk @piyushrpt - Brett, do not use the GUNW name for date and time, use the source granule for date/time.

We can fix the GUNW but that will not solve the problem.

There will be exactly one SLC in reference and secondary that is correct. That will be sufficient.

Here is the issue above using the first failed job.

gunw_path = 'S1-GUNW-D-R-048-tops-20230106_20221214-000000-00090E_00040N-PP-c254-v2_0_6.nc'

The first date is `20230106` but the _center time_ is in the next day.

 `S1A_IW_SLC__1SDV_20230106T235934_20230107T000001_046670_059804_D8E3`

Use the SLC ids _together_ to figure out time.

https://github.com/dbekaert/RAiDER/blob/dev/tools/RAiDER/aria/prepFromGUNW.py#L65-L75

**Comment by cmarshak** [comment]
For everyone's knowledge, I updated the Hyp3 Plugin that generates gunw so the final id of this product would be something like `'S1-GUNW-D-R-048-tops-20230106_20221214-235959-00090E_00040N-PP-c254-v2_0_6'`.

Here is the PR: https://github.com/ACCESS-Cloud-Based-InSAR/DockerizedTopsApp/pull/112

**Comment by bbuzz31** [comment]
Fixed in #500
